api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -61,7 +61,13 @@ public void validate(Object target, Errors errors) {
 		ValidationUtils.rejectIfEmpty(errors, "patient", "allergyapi.patient.required");
 		
 		Allergy allergy = (Allergy) target;
-		
+
+		if(allergy.getReactionNonCoded() != null) {
```

Comment:
thanks @Muta-Jonathan   have you tried these changes on a running instance of openmrs